Packages/com.unity.inputsystem/InputSystem/Plugins/UI/InputSystemUIInputModule.cs

```diff
@@ -1406,16 +1406,16 @@ private bool HasNoActions()
 
         private bool IsAnyActionEnabled()
         {
-            return (m_PointAction?.action?.enabled ?? true) &&
-                (m_LeftClickAction?.action?.enabled ?? true) &&
-                (m_RightClickAction?.action?.enabled ?? true) &&
-                (m_MiddleClickAction?.action?.enabled ?? true) &&
-                (m_MoveAction?.action?.enabled ?? true) &&
-                (m_SubmitAction?.action?.enabled ?? true) &&
-                (m_CancelAction?.action?.enabled ?? true) &&
-                (m_ScrollWheelAction?.action?.enabled ?? true) &&
-                (m_TrackedDeviceOrientationAction?.action?.enabled ?? true) &&
-                (m_TrackedDevicePositionAction?.action?.enabled ?? true);
+            return (m_PointAction?.action?.enabled ?? false) &&
+                (m_LeftClickAction?.action?.enabled ?? false) &&
+                (m_RightClickAction?.action?.enabled ?? false) &&
+                (m_MiddleClickAction?.action?.enabled ?? false) &&
+                (m_MoveAction?.action?.enabled ?? false) &&
+                (m_SubmitAction?.action?.enabled ?? false) &&
+                (m_CancelAction?.action?.enabled ?? false) &&
+                (m_ScrollWheelAction?.action?.enabled ?? false) &&
+                (m_TrackedDeviceOrientationAction?.action?.enabled ?? false) &&
+                (m_TrackedDevicePositionAction?.action?.enabled ?? false);
```

Comment:
Hmmm, this was possibly correct before, but the name of the method was wrong. The logic of "if any action is unassigned when I set a new asset on the input module, but all other actions are enabled, enable all actions" seems better than the alternative. Maybe the method could be renamed to IsAnyActionEnabledOrUnassigned?

Reply:
Yup, you're right. Classic mistake of reading the booleans wrong. Still the previous version was wrong, too. If you have *no* actions at all, it means it *will* automatically enable actions when you assign some.

I think this should read

```
            return (m_PointAction?.action?.enabled ?? false) ||
                (m_LeftClickAction?.action?.enabled ?? false) ||
                (m_RightClickAction?.action?.enabled ?? false) ||
                (m_MiddleClickAction?.action?.enabled ?? false) ||
                (m_MoveAction?.action?.enabled ?? false) ||
                (m_SubmitAction?.action?.enabled ?? false) ||
                (m_CancelAction?.action?.enabled ?? false) ||
                (m_ScrollWheelAction?.action?.enabled ?? false) ||
                (m_TrackedDeviceOrientationAction?.action?.enabled ?? false) ||
                (m_TrackedDevicePositionAction?.action?.enabled ?? false);
```

Reply:
Removed the method altogether and changed the logic of the `actionsAsset` setter.